pkg/dhclient/dhcp4.go

```diff
@@ -102,11 +105,19 @@ func (p *Packet4) Lease() *net.IPNet {
 	}
 }
 
+var (
+	ErrNoBootFile       = errors.New("no boot file name present in DHCP message")
+	ErrNoServerHostName = errors.New("no server host name present in DHCP message")
+)
+
 // Boot returns the boot file assigned.
 func (p *Packet4) Boot() (*url.URL, error) {
 	// TODO: This is not 100% right -- if a certain option is set, this
 	// stuff is encoded in options instead of in the packet's BootFile and
 	// ServerName fields.
+	if len(p.P.BootFileName) == 0 {
```

Comment:
Wait, but sometimes dhcp is used not in the context of netbooting. Wouldn't this break it for the general use case?

Reply:
The policy we prefer that if no valid boot path exists, that the system wait some minimum + random amount of time, then reboot.

In the case of DHCP failures, this gives DC Ops/automated systems, the chance to update services that will allow boot later. In the case of the loose cables or a system being put into service for the first time, or just having had repairs, it allows for power to be applied, *then* cables to be attached without having to send a person to hit a reset button, etc.

Reply:
This is just a library.

If you're just a `dhclient` that doesn't want to boot, don't call `Boot()`.

If you're a PXE client that *does* want to boot, then returning an error when we don't have information is the right way to go.